pkg/monitor/monitor.go

```diff
@@ -148,6 +150,24 @@ func (m *Monitor) PollNewFiles(ctx *context.Context, duration time.Duration) {
 				log.Infof("File %s already dispatched, skipping", file.Path)
 				continue
 			}
+			log.Infof("Downloading file %s to shared folder", file.Path)
+			basePath := m.Config.Monitor.BaseTmpDir
+			if basePath == "" {
+				basePath = "/tmp"
+			}
+			log.Debugf("Using temporary base path: %s", basePath)
+			fileEntry, err := m.FilesClient.Download(&file, basePath)
+			if err != nil {
+				log.Errorf("Failed to download %s: %s", file.Path, err)
+			}
+			log.Infof("Downloaded file to %s", fileEntry.Path)
+			if _, err := os.Stat(basePath); os.IsNotExist(err) {
+				log.Debugf("Temporary base path '%s' doesn't exist - creating", basePath)
+				if err = os.MkdirAll(basePath, 0755); err != nil {
+					log.Errorf("Cannot create temporary base path: %s", err.Error())
+				}
+			}
```

Comment:
Ok, one more question. Here the directory is getting creating after the download, should that be before?

Reply:
Yes, that's a good point.